mm2src/kdf_walletconnect/src/session/mod.rs

```diff
@@ -235,7 +234,7 @@ impl SessionManager {
     /// If a session with the same topic already exists, it will be overwritten.
     pub(crate) async fn add_session(&self, session: Session) {
         // set active session topic.
-        *self.0.active_topic.lock().await = Some(session.topic.clone());
+        *self.0.active_topic.lock().unwrap() = Some(session.topic.clone());
```

Comment:
such funcs were made async exclusively because of the async lock. we can make them sync now.